### PR TITLE
Label issues and PRs on open with 'needs triage'

### PR DESCRIPTION
**Pull request by jsignell**
This PR will add the 'needs triage' label to every PR and issue when it is opened (or reopened for issues). @jrbourbeau, @pavithraes, @scharlottej13, and I decided that this would be interesting to try. 

The idea is that this label will be automatically added, and then once someone responses to the issue, they should remove the label.

### CODE REVIEW

File: .github/workflows/label-all.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  pull_request_target:
+    types: [opened]
+  issues:
+    types: [opened, reopened]
+jobs:
+  label-all-on-open:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: andymckay/labeler@main
```
Comment (jrbourbeau) [inline .github/workflows/label-all.yml:11]: Can we use a specific release instead of pointing to `main`?

File: .github/workflows/label-all.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: "Issue and PR Labeler"
+on:
+  pull_request_target:
```
Comment (jrbourbeau) [inline .github/workflows/label-all.yml:3]: This is the first time I'm seeing us use `pull_request_target`. Is there a reason we can't use `pull_request` here instead?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell, apologies for the delayed review. I'm a bit confused on why we need both a `.github/workflows/label-prs.yml` and `.github/workflows/label-all.yml` file.

File: .github/workflows/label-all.yml (hunk 1 above)
Comment (jsignell) [reply .github/workflows/label-all.yml:11]: sure!

File: .github/workflows/label-all.yml (hunk 2 above)
Comment (jsignell) [reply .github/workflows/label-all.yml:3]: yeah I just copied this from the labeler one, but you are right. `pull_request` would also be fine.

**Pull request merged**